Getting started with Python (#1792)
microsoft/kiota: docs/get-started/python.md

---
parent: Get started
---

# Build SDKs for Python

## Required tools

- [Python 3.6+](https://www.python.org/)
- [pip 20.0+](https://pip.pypa.io/en/stable/)
- [Asyncio/any other supported async envronment e.g AnyIO, Trio.](https://docs.python.org/3/library/asyncio.html)

## Target project requirements

> Before you can compile and run the target project, you will need to initialize it. After initializing the test project, you will need to add references to the [abstraction](https://github.com/microsoft/kiota-abstractions-python), [authentication](https://github.com/microsoft/kiota-authentication-azure-python), [http](https://github.com/microsoft/kiota-http-python), [serialization JSON](https://github.com/microsoft/kiota-serialization-json-python), and [serialization Text](https://github.com/microsoft/kiota-serialization-text-python) packages.

Reviewer: ```suggestion
Before you can compile and run the target project, you will need to initialize it. After initializing the test project, you will need to add references to the [abstractions](https://github.com/microsoft/kiota-abstractions-python), [authentication](https://github.com/microsoft/kiota-authentication-azure-python), [http](https://github.com/microsoft/kiota-http-python), [serialization JSON](https://github.com/microsoft/kiota-serialization-json-python), and [serialization Text](https://github.com/microsoft/kiota-serialization-text-python) packages.
```

## Creating target projects

> **Note:** you can use an existing project if you have one, in that case, you can skip the following section.

Create a directory that will contain the new project.

## Adding dependencies
In your project directory, run the following commands on the terminal to install required dependencies
using `pip`:

```bash
pip install microsoft-kiota-abstractions
pip install microsoft-kiota-authentication-azure
pip install azure-identity
pip install microsoft-kiota-serialization-json
pip install microsoft-kiota-serialization-text
pip install microsoft-kiota-http
```

> **Note:** It is recommended to use a package manager/virtual environment to avoid installing packages
system wide. Read more [here](https://packaging.python.org/en/latest/).

> Only the first package, `microsoft-kiota-abstractions`, is required. The other packages provide default implementations that you can choose to replace with your own implementations if you wish.

Reviewer: ```suggestion
Only `microsoft-kiota-abstractions` is required to build the generated result. The other packages provide default implementations needed at runtime that you can choose to replace with your own implementations if you wish.
```

## Generating the SDK

Kiota generates SDKs from OpenAPI documents. Create a file named **getme.yml** and add the contents of the [Sample OpenAPI description](reference-openapi.md).

You can then use the Kiota command line tool to generate the SDK classes.

```shell

> kiota -l python -d ../getme.yml -c GetUserApiClient -n getuser/client -o ./client

Reviewer: ```suggestion
kiota generate -l python -d ../getme.yml -c GetUserApiClient -n getuser/client -o ./client
```

once #1864 is merged

---
parent: Get started
---

# Build SDKs for Python

## Required tools

> - [Python 3.10.4](https://www.python.org/)

Reviewer 2: Does it have to be these specific versions, or could you say Python 3.x?, pip 20.x?

Reviewer 3: Updated in https://github.com/microsoft/kiota/pull/1792/commits/3b79f65263a4910ff6329073bfa23b4b1dd5ff7a

- [pip 20.0.2](https://pip.pypa.io/en/stable/)

## Target project requirements

Before you can compile and run the target project, you will need to initialize it. After initializing the test project, you will need to add references to the [abstraction](https://github.com/microsoft/kiota-abstractions-python), [authentication](https://github.com/microsoft/kiota-authentication-azure-python), [http](https://github.com/microsoft/kiota-http-python), [serialization JSON](https://github.com/microsoft/kiota-serialization-json-python), and [serialization Text](https://github.com/microsoft/kiota-serialization-text-python) packages.

## Creating target projects

> **Note:** you can use an existing project if you have one, in that case, you can skip the following section.

Execute the following commands in the directory where you want to create a new project.

```bash

> py someCommand

Reviewer 2: What is this `py`? It doesn't work on my machine. Is it an alias for python or python3? We should change these commands to something that will work for any environment with just the pre-reqs.

Reviewer 3: Also updated in https://github.com/microsoft/kiota/pull/1792/commits/3b79f65263a4910ff6329073bfa23b4b1dd5ff7a

```

## Adding dependencies

Create a file named **getuser.py** and add the following code.

```bash

> py get github.com/microsoft/kiota-abstractions-python
> py get github.com/microsoft/kiota-http-python
> py get github.com/microsoft/kiota-serialization-json-python
> py get github.com/microsoft/kiota-serialization-text-python
> py get github.com/microsoft/kiota-authentication-azure-python
> py get github.com/Azure/azure-sdk-for-python/sdk/azidentity

Reviewer: ```suggestion
py get github.com/microsoft/microsoft-kiota-abstractions-python
py get github.com/microsoft/microsoft-kiota-http-python
py get github.com/microsoft/microsoft-kiota-serialization-json-python
py get github.com/microsoft/microsoft-kiota-serialization-text-python
py get github.com/microsoft/kiota-authentication-azure-python
py get github.com/Azure/azure-sdk-for-python/sdk/azidentity
```

> py get github.com/microsoft/microsoft-kiota-abstractions-python
> py get github.com/microsoft/microsoft-kiota-http-python
> py get github.com/microsoft/microsoft-kiota-serialization-json-python
> py get github.com/microsoft/microsoft-kiota-serialization-text-python
> py get github.com/microsoft/kiota-authentication-azure-python
> py get github.com/Azure/azure-sdk-for-python/sdk/azidentity

Reviewer: ```suggestion
py get microsoft-kiota-abstractions
py get microsoft-kiota-http
py get microsoft-kiota-serialization-json
py get microsoft-kiota-serialization-text
py get microsoft-kiota-authentication-azure
py get azure-identity
```

```

> Only the first package, `github.com/microsoft/kiota-abstractions-python`, is required. The other packages provide default implementations that you can choose to replace with your own implementations if you wish.

Reviewer: ```suggestion
Only the first package, `github.com/microsoft/microsoft-kiota-abstractions-python`, is required. The other packages provide default implementations that you can choose to replace with your own implementations if you wish.
```

> Only the first package, `github.com/microsoft/microsoft-kiota-abstractions-python`, is required. The other packages provide default implementations that you can choose to replace with your own implementations if you wish.

Reviewer: ```suggestion
Only the first package, `microsoft-kiota-abstractions`, is required. The other packages provide default implementations that you can choose to replace with your own implementations if you wish.
```

## Generating the SDK

Kiota generates SDKs from OpenAPI documents. Create a file named **getme.yml** and add the contents of the [Sample OpenAPI description](reference-openapi.md).

You can then use the Kiota command line tool to generate the SDK classes.

```shell
kiota -l py -d ../getme.yml -c GraphApiClient -n getuser/client -o ./client
```

## Creating an application registration

> **Note:** this step is required if your client will be calling APIs that are protected by the Microsoft Identity Platform like Microsoft Graph.

Follow the instructions in [Register an application for Microsoft identity platform authentication](register-app.md) to get an application ID (also know as a client ID).

## Creating the client application

Create a file in the root of the project named **getuser.py** and add the following code. Replace `YOUR_CLIENT_ID` with the client ID from your app registration.

```python

> code goes here

Reviewer: missing code sample

Reviewer 3: Added in https://github.com/microsoft/kiota/pull/1792/commits/3b79f65263a4910ff6329073bfa23b4b1dd5ff7a
